apps/web/pages/api/integrations/slackapp/listener/[...args].ts

```diff
@@ -0,0 +1,78 @@
+import { createHmac } from "crypto";
+import dayjs from "dayjs";
+import { NextApiRequest, NextApiResponse } from "next";
+import { stringify } from "querystring";
+
+import appStore from "@calcom/app-store";
+
+import { HttpError } from "@lib/core/http/error";
+
+const signingSecret = process.env.SLACK_SIGNING_SECRET;
+
+function verifyHash(req: NextApiRequest, res: NextApiResponse) {
+  const body = req.body;
+  const timeStamp = req.headers["x-slack-request-timestamp"] as string; // Always returns a string and not a string[]
+  const slackSignature = req.headers["x-slack-signature"] as string;
+  const currentTime = dayjs().unix();
+
+  // Ensure that there is a timestamp
+  if (!timeStamp) {
+    return res.status(400).json({ message: "Missing X-Slack-Request-Timestamp header" });
+  }
+
+  // If there is no signging secret in env there is no way to check that the request is valid.
+  if (!signingSecret) {
+    return res.status(400).json({ message: "Missing process.env.SLACK_SIGNING_SECRET" });
+  }
+
+  // Ensure that the timestamp is within the last 5 minutes (300 seconds)
+  if (Math.abs(currentTime - parseInt(timeStamp)) > 60 * 5) {
+    return res.status(400).json({ message: "Request is too old" });
+  }
+
+  const hmac = createHmac("sha256", signingSecret);
+  const [version, hash] = slackSignature.split("=");
+
+  const payload = body.payload && JSON.parse(body.payload);
+  console.log(payload);
+
+  hmac.update(`${version}:${timeStamp}:${stringify(payload ? { payload } : body)}`);
```

Comment:
See: https://api.slack.com/authentication/verifying-requests-from-slack for the hashing side of this verification